backend/tests/test_oauth.py

```
@@ -1576,6 +1577,790 @@ def test_callback_unknown_state_redirects_with_error():
+
+    # A value is matched as a COMPLETE token: a prefix match would leave the tail of the
+    # credential visible next to a "<redacted>" claiming otherwise. Provider secrets may
+    # legally contain spaces or escaped quotes.
+    spaced = oauth_flow.redact_secrets('{"client_secret": "abc def ghi"}')
+    assert "abc" not in spaced and "ghi" not in spaced
+    assert spaced == '{"client_secret": "<redacted>"}'
+    escaped = oauth_flow.redact_secrets('{"client_secret":"abc\\"def"}')
+    assert "def" not in escaped, escaped
+    assert oauth_flow.redact_secrets("{'client_secret': 'a b'}") == "{'client_secret': '<redacted>'}"
+
+    token_body = "Token exchange failed (400): access_token=AT-live&refresh_token=RT-live&scope=read"
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "token_body"

(S105)